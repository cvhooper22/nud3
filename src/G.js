import React from 'react';
import splitSVGElementProps from './helpers/splitSVGElementProps';

export default function G (props) {
  const [elementProps, newProps] = splitSVGElementProps(props);

  const children = React.Children.map(props.children, (node) => {
    return React.cloneElement(node, { ...newProps, ...node.props });
  });

  return (
    <g { ...elementProps }>
      { children }
    </g>
  );
}
G.propTypes = {
  children: React.PropTypes.node,
};
