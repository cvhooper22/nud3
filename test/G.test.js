import React from 'react';
import renderer from 'react-test-renderer';
import G from '../src/G';


describe('G.js', () => {
  it('should take proper SVG items and pass', () => {
    const Fake = (props) => {
      return <path mask={ props.notRealSVG } />;
    };
    const tree = renderer.create(
      <G mask="123" notRealSVG={ 436 }>
        <Fake />
      </G>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
