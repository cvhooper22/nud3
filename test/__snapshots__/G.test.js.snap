exports[`G.js should take proper SVG items and pass 1`] = `
<g
  mask="123">
  <path
    mask={436} />
</g>
`;
